### PR TITLE
tests and code - all tests pass

### PR DESCRIPTION
**Pull request by bmraubo**
Tests for custom delimiters, including edge cases. Code refactored through addition of Empty func to limit the reuse of code.

### CODE REVIEW

File: kata.py [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+                        answer += int(num)
+                    return answer
+            elif Empty(string.split('\n')[1]):
+                return 0
+            else:
+                answer = 0
+                delimiter = string.split('\n')[0].strip('//')
+                for num in string.split('\n')[1].split(delimiter):
+                    answer += int(num)
+                return answer
```
Comment (martingaston) [inline kata.py:24]: Nested `if` conditionals are often a great indicator that your function is starting to do many things - which can create confusion, and code that's very hard to test and change (the Gilded Rose is perhaps the idiomatic example of that). I can count 9 different possible paths of execution for this function now - all of which can be opportunities to bugs to worm their way in. 

This is probably the best/first opportunity in the kata to really think about the 'refactor' stage of red/green/refactor - notice how your original code just sort of stays sitting there quite merrily at the end of all those if/elses, with a little extra indentation.

Might your function now have multiple responsibilities? `Add` is doing quite a bit more than just adding - and while we're constrained a little by the fixed method signature perhaps we can extract some additional functions out here to keep everything having one single responsibility.

I think it's probably fair to say that in an ideal world our code in this function wouldn't get much more complicated than the original contents in that final `else` block.

File: test_kata.py [hunk 2]
```diff
@@ -3,23 +3,42 @@
...
         self.assertEqual(Add('1\n3'), 4)
 
     def test_new_line_comma(self):
         self.assertEqual(Add('1\n3,5'), 9)
 
+    #Step 4 tests
+    def test_other_delimiters(self):
+        self.assertEqual(Add('//&\n4&9'), 13)
+
+    def test_howdy_delimiter(self):
```
Comment (martingaston) [inline test_kata.py:31]: `test_howdy_delimiter` or `test_a_delimiter_can_have_multiple_characters`? 

Imagine you go away for this for six months, come back, and have to remember the significance of `howdy` 😄
